sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1571,6 +1659,184 @@ a normal SYCL command-group submission.
 associated with the graph node resulting from this command-group submission is
 different from the one with which the dynamic_parameter was created.
 
+|
+[source,c++]
+----
+void handler::ext_oneapi_barrier(const std::vector<dynamic_event>& waitList);
+----
+
+|Overload of `ext_oneapi_barrier` that takes a list of dynamic events to wait
+on.
+
+This function has the same semantics as `ext_oneapi_barrier(const
+std::vector<event>&)`.
+
+Parameters:
+
+* `waitList` - List of dynamic event dependencies for this barrier.
+
+Exceptions:
+
+* Throws synchronously with error code `invalid` if this function is called from
+a normal SYCL command-group submission.
+
+* Throws synchronously with error code `invalid` if any of the `sycl::event`s
+associated with `waitList` came from the same graph that the graph node
+resulting from this command-group submission is associated with.
+
+|
+[source,c++]
+----
+void handler::depends_on(dynamic_event depEvent);
+----
+
+|Overload of `depends_on()` which takes a dynamic event dependency.
+
+This function has the same semantics as `depends_on(event)`.
+
+Parameters:
+
+* `depEvent` - Dynamic event dependency.
+
+Exceptions:
```

Comment:
Did we consider whether it should be an error if the dynamic event isn't backed by a SYCL event at this point? This affects the UR design in that otherwise we have to provide a way to say that although a command has an empty event wait-list now, it can be updated in future. (not that this is a big deal, just whether it's a use case we expect).